### ISSUE TITLE
Improve resource timeline performance with virtual rendering

### ISSUE THREAD

**Issue by AndrejJurkin** [fullcalendar/fullcalendar#5673]
### Checklist

Please mark these items with an [x]

  - [X] I've already searched through [existing tickets](https://fullcalendar.io/issues)
  - [X] Other people will find this feature useful

### Feature Description

Full callendar is struggling with performance when rendering a big dataset in Resource Timeline view.

Let's say I need to render 5000 rows of resources as well as the events. In the current state, the UI is just unresponsive and very laggy.

A very straightforward way to improve this would be virtual rendering. Using [https://github.com/bvaughn/react-virtualized](https://github.com/bvaughn/react-virtualized) for example.

Here's an example of a library utilizing virtual rendering - [http://bhp-react-timeline-9k.s3-website-ap-southeast-2.amazonaws.com/](http://bhp-react-timeline-9k.s3-website-ap-southeast-2.amazonaws.com/)

Do you think something like this would be possible here?

I'm also open to implement this feature.

Thanks!

**Comment by harryy2510** [comment]
Hello team,
Do we have any plans for this?
I was going though a few ideas if we plan for this

Please have a look at this

http://bvaughn.github.io/react-virtualized/#/components/MultiGrid

**Comment by gablabelle** [comment]
> ### Checklist
> Please mark these items with an [x]
> 
> * [x]  I've already searched through [existing tickets](https://fullcalendar.io/issues)
> * [x]  Other people will find this feature useful
> 
> ### Feature Description
> Full callendar is struggling with performance when rendering a big dataset in Resource Timeline view.
> 
> Let's say I need to render 5000 rows of resources as well as the events. In the current state, the UI is just unresponsive and very laggy.
> 
> A very straightforward way to improve this would be virtual rendering. Using https://github.com/bvaughn/react-virtualized for example.
> 
> Here's an example of a library utilizing virtual rendering - http://bhp-react-timeline-9k.s3-website-ap-southeast-2.amazonaws.com/
> 
> Do you think something like this would be possible here?
> 
> I'm also open to implement this feature.
> 
> Thanks!

Same here ... 10000+ events make the calendar unusable. It really needs to be virtualized. @AndrejJurkin May I ask what library is being used in that demo you've linked?

**Comment by sqpchris** [comment]
This would be very useful for some of my projects. The resource timeline does everything we need, however the performance issues when using a large number of events means that we've been unable to use it for some projects.

**Comment by someguythatcodes** [comment]
This would be very useful for some of my projects. The resource timeline does everything we need, however the performance issues when using a large number of events means that we've been unable to use it for some projects.

**Comment by apbarratt** [comment]
I can confirm that this is the feature we are having major problems with and I am now moving toward the option of recommending we find a non-fullcalendar solution given that there appears to be no intention of movement on repairing this flaw.  I've found other gantt chart libraries that specifically call out hight performance with large data sets as a feature and whilst I'd really rather not have to spend such a significant amount of resources detangling full calendar from our product, I'm having to conclude that the lack of support shown in the six year old ticket (#3003), that we would be foolish to allow this product to become further entwined with ours.

**Comment by AngelZacarias** [comment]
I see this was moved to the v7 milestone... and on the [roadmap](https://fullcalendar.io/roadmap) is listed (I think so) as "row/col virtualization" under Q4... So could I expect to have this new feature on the v7 at the end of the year? If not, is there some workaround that can be applied? 🤔

**Comment by lpotapczuk** [comment]
I can see that it now has been shifted to Q1/Q2 24 :/ That is a bummer, since it is the only thing that is actually causing me to switch the library due to the rendering performance... Did anyone found any workaround on this? Perhaps any forks?

**Comment by AngelZacarias** [comment]
Btw, there are more discussion about this kind of things on #6478 and #3003, I wouldn't say those are duplications but seems like it is a really desired and expected enhancement 👀

**Comment by lpotapczuk** [comment]
Exactly. @arshaw if possible, I would like to help there anyhow, but I have no idea how to connect with you on this work :)

**Comment by VisileanRushiC** [comment]
Hello @AndrejJurkin,

🚨 I'm also facing this issue and it's been quite challenging. It's causing me to consider switching libraries due to rendering performance concerns. However, I've decided to stick with it because v7 is just around the corner and I'm hopeful for improvements. 🌟

Has anyone found any workaround or tips to mitigate this in the meantime? Any advice would be greatly appreciated!

**Comment by arshaw** [comment]
This has been improved in [v7.0.0-rc.0](https://github.com/fullcalendar/fullcalendar-workspace/releases/tag/v7.0.0-rc.0), please see [this comment in another issue](https://github.com/fullcalendar/fullcalendar/issues/3003) for more details.

**Comment by arshaw** [comment]
I've snuck this into the v7 milestone and released a beta. Paid customers were demanding it an it couldn't wait until v7.1

https://github.com/fullcalendar/fullcalendar-workspace/releases/tag/v7.0.0-beta.6 (includes demos)

I'm not exactly sure when a v7 release candidate will be ready, but you can track progress here: https://github.com/fullcalendar/fullcalendar/issues/7907
